addons/stock/models/stock_move.py

```diff
@@ -830,7 +838,7 @@ def _merge_moves(self, merge_into=False):
         for candidate_moves in candidate_moves_list:
             # First step find move to merge.
             candidate_moves = candidate_moves.with_context(prefetch_fields=False)
-            for k, g in groupby(sorted(candidate_moves, key=self._prepare_merge_move_sort_method), key=itemgetter(*distinct_fields)):
+            for __, g in groupby(sorted(candidate_moves, key=self._prepare_merge_move_sort_method, reverse=True), key=itemgetter(*distinct_fields)):
```

Comment:
I don't get it, why reversing? The merge should work in any order :o:

Reply:
It's related to the created_production_id. When the negative move is merged, it doesn't have a created_production_id yet, but if those moves aren't reversed, the old move will be merged into the negative one, which will then lose the old created_production_id (it will be overidden by the False created_production_id)